sklearn/metrics/tests/test_pairwise.py

```diff
@@ -407,6 +407,28 @@ def test_euclidean_distances():
     assert_greater(np.max(np.abs(wrong_D - D1)), .01)
 
 
+def test_cosine_distances():
+    # Check the pairwise cosine_distances distances computation
+    rng = np.random.RandomState(1337)
+    x = np.abs(rng.rand(910))
+    XA = np.vstack([x, x])
+    D = cosine_distances(XA)
+    assert_array_almost_equal(D, [[0., 0.], [0., 0.]])
+    # check that all elements are in [0, 2]
+    assert_true(np.all(D >= 0.))
```

Comment:
This seems pretty vacuous after the test that all entries are zero? Maybe tests these with a larger random matrix?